_posts/2023-12-10-Getting Started with Rust SDK.md

```diff
@@ -0,0 +1,326 @@
+
+---
+
+We provide a Rust SDK for committing transactions to create and transfer assets. Despite NexRes being written in C++, users can interface with ResilientDB through Rust. This article assumes you have setup a ResilientDB instance that is accessible through HTTP requests and are familiar with Rust, we will walk you through basics of using ResilientDB in your first Rust project.
+
+# Prerequisites
+### Install Rust and Cargo 
+The easiest way to get Cargo is to install the current stable release of Rust by using rustup. Installing Rust using rustup will also install cargo.
+On Linux and macOS systems, this is done as follows:
+
+> `curl https://sh.rustup.rs -sSf | sh`
+
+It will download a script, and start the installation. If everything goes well, you’ll see this appear:
+
+> `Rust is installed now. Great!`
+
+On Windows, download and run `rustup-init.exe`. It will start the installation in a console and present the above message on success.
+
+After this, you can use the rustup command to also install beta or nightly channels for Rust and Cargo.
+
+You can find more informatin on rust on [The Cargo Book](https://doc.rust-lang.org/cargo/getting-started/installation.html).
+
+### Starting a new Rust Project and installing the SDK
+You can start a new rust project through cargo with the following command: 
+> `cargo new resilientDB_test`
+
+You should now see a new folder generated named `resilientDB_test`.
+```
+$ cd resilientDB_test
+$ tree .
+.
+├── Cargo.toml
+└── src
+    └── main.rs
+
+1 directory, 2 files
+``` 
+You can get started with the Rust SDK in one of two ways: 
+1. You can simply run the following command:
+   ```
+   cargo add resilientdb_rust_sdk
+   ```
+2. You can also add the following line to your `Cargo.toml`
+   ```
+   resilientdb_rust_sdk = "0.1.1"
+   ```
+
+It's that easy! You are now good to go with our SDK. Make sure you have a ResilientDB instance setup, for more info on that follow this [blog](https://blog.resilientdb.com/2023/11/22/Deploying-ResilientDB.html). This tutorial also makes a graphql query to resdb, if you don't have graphql setup take a look at this other [blog]().
+
+### Dependencies
+
+You may need to install the following dependencies as our library has `future` return types, meaning some returns from our APIs would need to be unwrapped within an `async` function. 
+
+This is by design as we encourage users to handle errors on their end as many of the calls are made asynchrounously. Following good error handling practices would be beneficial to developing a secure application with ResilientDB. You can run obtain the dependencies through: 
+
+1. Running the following command
+   > `cargo add tokio serde serde_json reqwest`
+   
+2. Adding this to your Cargo.toml file:
+   ```rust
+    tokio = { version = "1", features = ["full"] }
+    serde = { version = "1.0", features = ["derive"] }
+    serde_json = "1.0"
+    reqwest = {version = "0.11.22", features = ["blocking", "json"]}
+   ```
```

Comment:
When I follow your steps on 
 ```
 cargo add resilientdb_rust_sdk
 ```
 
 This setting is already there and it is a little different from your description. 
 Any version number is missing?
 
 ```
 [dependencies]
 reqwest = "0.11.22"
 resilientdb_rust_sdk = "0.1.2"
 serde = "1.0.193"
 serde_json = "1.0.108"
 tokio = "1.35.0"
 ```

Reply:
Yeah I pushed v0.1.2, so that should be the latest version. The Cargo.toml that you have sent looks correct to me.

Reply:
> Yeah I pushed v0.1.2, so that should be the latest version. The Cargo.toml that you have sent looks correct to me.

What I provided is the default setting.
So it would be good to explain this situation. Like, if xxx is not there, please add xxx.
Otherwise, it will be a little confusing.

Reply:
I agree, it would be simpler if clarified. `cargo add ResilientDB_rust_sdk` will always add the latest version unless defined. Currently the latest version happens to be v0.1.2.

Reply:
Could you please modify it in this blog by creating another pull request?

Reply:
Will do